types/jest/index.d.ts

```diff
@@ -820,7 +875,7 @@ declare namespace jest {
          *
          * expect(houseForSale).toHaveProperty('kitchen.area', 20);
          */
-        toHaveProperty(propertyPath: string | any[], value?: any): R;
+        toHaveProperty<E>(propertyPath: string | any[], value?: E): R;
```

Comment:
```suggestion
        toHaveProperty<E = any>(propertyPath: string | any[], value?: E): R;
```

Explicit fallback rather than implicit